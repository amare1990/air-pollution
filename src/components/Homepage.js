import React from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import '../styles/Homepage.css';

const Homepage = () => {
  const navigate = useNavigate();
  const cityData = useSelector((state) => state.city.data);
  const cityName = cityData.name;

  if (cityName === undefined) {
    return null;
  }

  const handleDetailsClick = () => {
    const path = '/Details';
    navigate(path);
  };

  return (
    <div className="city-categ">
      <p className="city-name-categ">{cityName}</p>
      <div className="city-data-categ">
        <p>
          Lat:
          {' '}
          {'      '}
          <span className="geo lat">{cityData.lat}</span>
          {'   ' }
        </p>
        <p>
          Lon:
          {'     '}
          <span className="geo lon">{cityData.lon}</span>
        </p>
      </div>
      <button
        type="button"
        className="btn btn-categ"
        onClick={handleDetailsClick}
      >
        See the details
      </button>

    </div>
  );
};

export default Homepage;
