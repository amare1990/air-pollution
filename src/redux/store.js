import { configureStore } from '@reduxjs/toolkit';
import pollutionReducer from './current/current';
import cityReducer from './city/city';

export default configureStore({
  reducer: {
    pollution: pollutionReducer,
    city: cityReducer,
  },
});
