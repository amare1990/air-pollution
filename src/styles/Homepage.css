.city-categ {
  background-color: rgb(170, 19, 102);
  opacity: 0.89;
}

.city-name-categ {
  color: #fff;
  font-size: 24px;
}

.city-data-categ {
  font-size: 14px;
  color: #fff;
}

.btn-categ {
  padding: 5px 7px;
  margin-bottom: 5px;
  font-size: 16px;
  border-radius: 7px;
  cursor: pointer;
}

.btn-categ:hover {
  background-color: aqua;
}
