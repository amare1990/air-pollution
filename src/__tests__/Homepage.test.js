import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Homepage from '../components/Homepage';
import store from '../redux/store';

test('Testing if SearchBar component renders correctly', () => {
  const homepageBarSnapshot = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(homepageBarSnapshot).toMatchSnapshot();
});
