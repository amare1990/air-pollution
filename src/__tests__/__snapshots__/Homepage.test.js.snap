// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if SearchBar component renders correctly 1`] = `null`;
